src/ixdat/config.py

```diff
@@ -181,20 +181,58 @@ def QUANT_DIRECTORY(self, quant_directory):
         if self._USE_QUANT:
             self._USE_QUANT = True  # gets the quant directory to be reset
 
-
 class _CinfData:
-    def __init__(
-        self,
-    ):
+
+    def __init__(self,):
         self.cinfdata = None
+        self._managed_cinfdata_object = None
+        self._context_manager_kwargs = None
+        print('initialised cinfdatabase connection')
 
     def activate(self):
+        print('activated cinfdatabase connection')
         from cinfdata import Cinfdata
 
         self.DB = Cinfdata
 
     def connect(self, setup_name=None, grouping_column=None):
+        print('connected cinfdatabase connection')
         return self.DB(setup_name=setup_name, grouping_column=grouping_column)
 
+    def __call__(self, **kwargs):
```

Comment:
Alright. You will need to clean this up a little bit. Update the docstrings, loose the prints in `__call__` and `__enter__` and `__exit__` since those will be hit on every use of the context manager.